iceoryx_binding_c/test/moduletests/test_subscriber.cpp

```diff
@@ -257,52 +257,51 @@ TEST_F(iox_sub_test, sendingTooMuchLeadsToLostChunks)
 
 TEST_F(iox_sub_test, attachingToWaitSetWorks)
 {
-    EXPECT_EQ(iox_sub_attach_to_waitset(m_sut, m_waitSet.get(), SubscriberEvent_HAS_NEW_SAMPLES, 0, NULL),
```

Comment:
We can already think about naming of events. `HAS_NEW_SAMPLES` currently means that the there are samples in the queue. They are new from the point of view that the user has not seen them. But when the user does not take them after a `waitset.wait()`, the trigger is still active even when the samples in the queue are new in the sense that they arrived after the last check

Maybe: `HAS_NEW_SAMPES` -> `HAS_QUEUED_SAMPLES`. This trigger is active whenever there are samples in the queue. But then we should also rename `hasNewSamples` in `hasQueuedSamples`

The other trigger we already discussed but did not introduce is the information whether new Samples arrived since the last check, no matter if there are some queued or not. This should be the `hasNewSamples` call and the `HAS_NEW_SAMPLES` event. 

Maybe it makes sense to clarify this now t avoid changing behavior of existing methods and events

Reply:
Event is called `HAS_SAMPLES` and the corresponding method call `hasSamples`